packages/plugin-commands-publishing/src/publish.ts

```diff
@@ -164,7 +164,20 @@ Do you want to continue?`,
           'prepack',
         ], publishManifest)
       }
-      const { status } = runNpm(opts.npmPath, ['publish', '--ignore-scripts', ...opts.argv.original.slice(1)])
+
+      const args = opts.argv.original.slice(1)
+      const index = args.indexOf('--publish-branch')
+      if (index !== -1) {
+        // If --publish-branch follows with another cli option, only remove this argument
+        // otherwise remove the following argument as well
+        if (args[index + 1] && args[index + 1].startsWith('--')) {
```

Comment:
what about short aliases of options? I think this will work better
```suggestion
        if (args[index + 1]?.startsWith('-')) {
```